.github/workflows/pulp_images.yml

```diff
@@ -94,107 +260,59 @@ jobs:
-
-            podman push ghcr.io/pulp/${APP}-minimal:$CURRENT_VERSION
-            podman push ghcr.io/pulp/${APP}-web:$CURRENT_VERSION
-          fi
-          echo $CURRENT_VERSION
       - name: Docker login
         if: github.event_name != 'pull_request'
         env:
           DOCKER_BOT_PASSWORD: ${{ secrets.DOCKER_BOT_PASSWORD }}
           DOCKER_BOT_USERNAME: ${{ secrets.DOCKER_BOT_USERNAME }}
         run: echo "$DOCKER_BOT_PASSWORD" | podman login -u "$DOCKER_BOT_USERNAME" --password-stdin docker.io
-      - name: Push images to dockerhub
-        if: github.event_name != 'pull_request'
-        run: |
-          podman tag pulp/base:latest docker.io/pulp/base:latest
-          podman tag pulp/${APP}-minimal:$IMAGE_TAG docker.io/pulp/${APP}-minimal:$IMAGE_TAG
-          podman tag pulp/${APP}-web:$IMAGE_TAG docker.io/pulp/${APP}-web:$IMAGE_TAG
-
-          podman push docker.io/pulp/base:latest
-          podman push docker.io/pulp/${APP}-minimal:$IMAGE_TAG
-          podman push docker.io/pulp/${APP}-web:$IMAGE_TAG
-
-          if [[ "$IMAGE_TAG" == "stable" ]]; then
-            podman tag pulp/${APP}-minimal:stable docker.io/pulp/${APP}-minimal:latest
-            podman tag pulp/${APP}-web:stable docker.io/pulp/${APP}-web:latest
-
-            podman push docker.io/pulp/${APP}-minimal:latest
-            podman push docker.io/pulp/${APP}-web:latest
-
-            CURRENT_VERSION=$(podman run pulp/${APP}-minimal:stable bash -c "pip3 show ${PIP_PKG_NAME_FOR_VERSION_STRING} | sed -n -e 's/Version: //p'")
-            echo $CURRENT_VERSION
-
-            podman tag pulp/${APP}-minimal:stable docker.io/pulp/${APP}-minimal:$CURRENT_VERSION
-            podman tag pulp/${APP}-web:stable docker.io/pulp/${APP}-web:$CURRENT_VERSION
-
-            podman push docker.io/pulp/${APP}-minimal:$CURRENT_VERSION
-            podman push docker.io/pulp/${APP}-web:$CURRENT_VERSION
-          fi
-          echo $CURRENT_VERSION
       - name: Quay login
         if: github.event_name != 'pull_request'
         env:
           QUAY_BOT_PASSWORD: ${{ secrets.QUAY_BOT_PASSWORD }}
           QUAY_BOT_USERNAME: ${{ secrets.QUAY_BOT_USERNAME }}
         run: echo "$QUAY_BOT_PASSWORD" | podman login -u "$QUAY_BOT_USERNAME" --password-stdin quay.io
-      - name: Push images to quay
+      - name: Push images to registries
         if: github.event_name != 'pull_request'
         run: |
-          podman tag pulp/base:latest quay.io/pulp/base:latest
-          podman tag pulp/${APP}-minimal:$IMAGE_TAG quay.io/pulp/${APP}-minimal:$IMAGE_TAG
-          podman tag pulp/${APP}-web:$IMAGE_TAG quay.io/pulp/${APP}-web:$IMAGE_TAG
+          for registry in gchr.io docker.io quay.io; do
 
-          podman push quay.io/pulp/base:latest
-          podman push quay.io/pulp/${APP}-minimal:$IMAGE_TAG
-          podman push quay.io/pulp/${APP}-web:$IMAGE_TAG
+            # Technically multiple jobs will build and push these 2 images
```

Comment:
wouldn't it be better to cache the images and then have another job restore them and push them once?

Reply:
It might take the overall release process take longer.

Because if we entirely separate the build jobs from the push jobs, then we have to restore/load the images repeatedly. The restore and load tasks (for the base images alone) take about 1 minute each, but restore/load the app images will take even longer.

Alternatively, we could do a single job to push the base images after the unified jobs to build/push the app images. This would probably be the best approach, it would add about 1 minute to runtime.

Reply:
Oh, also. The latter does is not so feasible. We rely on the app images to determine the tags for the base images. We could reduce which tags we publish for the base images, we could just use the tag of the branch. So currently, the tags would be "latest" (not "3.22"), "3.21", and older.

Reply:
Per our convo, I separated the steps to push the base images vs push the app images. We will see how long they take.